tests/contrib/requests/test_requests.py

```diff
@@ -302,6 +302,23 @@ def test_user_service_name_precedence(self):
         assert s.name == "requests.request"
         assert s.service == "clients"
 
+    def test_split_by_domain_with_email(self):
```

Comment:
does the email matter? or is it just the presence of a query string?

Reply:
It's the presence of `@` in the url. The previous parsing logic did not account for this case. With this change we will use `urllib.parse.urlparse(url)` to get the hostname so this should not be an issue

Reply:
can you include that context here and maybe in the PR description?